lib/debugger/debugger.dart

```diff
@@ -0,0 +1,1309 @@
+            scrollTo: new Pos(location.line - 1));
+      }
+    });
+
+    scriptsView = new ScriptsView(debuggerState.getShortScriptName);
+    scriptsView.onSelectionChanged.listen((ScriptRef scriptRef) async {
+      if (scriptRef == null) {
+        _displaySource(null);
+        return;
+      }
+
+      final IsolateRef isolateRef =
+          serviceManager.isolateManager.selectedIsolate;
+      final dynamic result =
+          await serviceManager.service.getObject(isolateRef.id, scriptRef.id);
+
+      if (result is Script) {
+        _displaySource(result);
+      } else {
+        _displaySource(null);
+      }
+    });
+
+    final PNavMenu menu = new PNavMenu(<CoreElement>[
+      new PNavMenuItem('Call stack')
+        ..click(() => callStackView.element.toggleAttribute('hidden')),
+      callStackView.element,
+      new PNavMenuItem('Variables')
+        ..click(() => variablesView.element.toggleAttribute('hidden')),
+      variablesView.element,
+      new PNavMenuItem('Breakpoints')
+        ..add(_breakpointsCountDiv)
+        ..click(() => breakpointsView.element.toggleAttribute('hidden')),
+      breakpointsView.element,
+      new PNavMenuItem('Scripts')
+        ..add(
+          _scriptCountDiv = span(text: '0', c: 'counter'),
+        )
+        ..click(() => scriptsView.element.toggleAttribute('hidden')),
+      scriptsView.element,
+    ], supportsSelection: false)
+      ..flex()
+      ..layoutVertical();
+
+    debuggerState.onBreakpointsChanged.listen((List<Breakpoint> breakpoints) {
+      breakpointsView.showBreakpoints(breakpoints);
+    });
+
+    return menu;
+  }
+
+  void _handleConnectionStart(VmService service) {
+    debuggerState.setVmService(serviceManager.service);
+
+    deviceStatus.element.text =
+        '${serviceManager.vm.targetCPU} ${serviceManager.vm.architectureBits}-bit';
+
+    service.onStdoutEvent.listen((Event e) {
+      final String message = decodeBase64(e.bytes);
+      consoleArea.append(message);
+    });
+
+    service.onStderrEvent.listen((Event e) {
+      final String message = decodeBase64(e.bytes);
+      consoleArea.append(message, isError: true);
+    });
+  }
+
+  void _handleIsolateChanged(IsolateRef isolateRef) {
+    if (isolateRef == null) {
+      scriptsView.clearScripts();
+      _scriptCountDiv.text = '0';
+
+      debuggerState.switchToIsolate(isolateRef);
+
+      return;
+    }
+
+    debuggerState.switchToIsolate(isolateRef);
+
+    serviceManager.service.getIsolate(isolateRef.id).then((dynamic result) {
+      if (result is Isolate) {
+        _populateFromIsolate(result);
+      } else {
+        scriptsView.clearScripts();
+        _scriptCountDiv.text = '0';
+      }
+    }).catchError((dynamic e) {
+      framework.showError('Error retrieving isolate information', e);
+    });
+  }
+
+  void _handleConnectionStop(dynamic event) {
+    deviceStatus.element.text = '';
+
+    scriptsView.clearScripts();
+    _scriptCountDiv.text = '0';
```

Comment:
I see setting the state of _scriptCountDiv a lot of places. Consider adding a helper that syncs its value to the current script count.